README.md

```diff
@@ -99,12 +114,44 @@ A
 BA
 CBA
 DCBA
+
+function letterTriangle(n) {
+var space = '';
+var alpha = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k','l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'];
+for (var m = 0; m < n; m++){
+for (var i = m; i>=0; i--) {
+space+=alpha[i];
+}
+space+='\n'; 
+}
+return space;
+}
 ```
 
 **b)**
 Adapting your solution to part a), write a function which returns a string containing the entire lyrics for the song "The Twelve Days of Christmas".  Make sure that your result is grammatically and typographically correct (include line breaks, commas, etc. where needed), but keep redundancy within your program to a minimum.  Don't just `console.log()` each line; as before, return them together as a single string which includes a '\n' at each line break.
 You may want to use helper functions, loops, and/or arrays to store repeated elements.
 
+var day = ['first', 'second', 'third', 'fourth', 'fifth', 'sixth', 'seventh', 'eigth', 'ninth', 'tenth', 'eleventh', 'twelfth'];
+var chorus = ['And a partridge in a pear tree', '2 turtle doves', '3 French hens', '4 calling birds', '5 gold rings', '6 geese a-laying', '7 swans a-swimming', '8 maids a-milking', '9 ladies dancing', '10 lords a-leaping', '11 pipers piping', '12 drummers drumming'];
+function generateSong(n) {
+var lyric = '';
+for (var i = 0; i < n; i++) {
+lyric += 'On the ' + day[i] + ' day of Christmas, \nmy truelove gave to me: \n';
+if (day[i] === 'first') {
+lyric += 'A partridge in a pear tree.' + '\n';
+} else {
+for (var m = i; m >= 0; m--) {
+lyric += chorus[m] + '\n';
+};
+};
+lyric += '\n';
+}
+return lyric;
+}
+
```

Comment:
This looks like it's pretty much right, and demonstrates the fact that you were able to construct a logical solution. For some reason, I couldn't get this one to run in the console, but that could very well be my fault. Nice work here!